langlib/langlib-test/src/test/resources/test-src/regexp_negative_test.bal

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2023 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/lang.regexp;
+
+public function testNegativeIndexFind() returns error? {
+    string:RegExp r = re `e`;
+    regexp:Span? _ = check (trap r.find("Hello", -3));
```

Comment:
Check where applicable
``` suggestion
    _ = check (trap r.find("Hello", -3));
```

Reply:
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/39238/commits/30d5dd4058fc549010a92ea234b4ede20d3998b4